bump up pytorch bin to 0714

Gasoonjia opened the pull request:
Summary: as title

Differential Revision: D78181215

swolchok commented:
test-setup-linux-gcc flaked, just clicked rerun for you

pianpwk reviewed (approved):
slice changes look good to me

Gasoonjia committed:
bump up pytorch pin to 0714 (#12407)

Summary:

pin bumped with updated test cases

Reviewed By: shoumikhin, pianpwk

Differential Revision: D78181215
